src/linux/ptrace_dumper.rs

```diff
@@ -634,39 +567,44 @@ impl PtraceDumper {
         mapping: &mut MappingInfo,
         pid: Pid,
     ) -> Result<Vec<u8>, DumperError> {
-        if !MappingInfo::is_mapped_file_safe_to_open(&mapping.name) {
-            return Err(DumperError::NotSafeToOpenMapping(
-                mapping.name.clone().unwrap_or_default(),
-            ));
-        }
-
-        // Special-case linux-gate because it's not a real file.
-        if mapping.name.as_deref() == Some(LINUX_GATE_LIBRARY_NAME.as_ref()) {
-            if pid == std::process::id().try_into()? {
-                let mem_slice = unsafe {
-                    std::slice::from_raw_parts(mapping.start_address as *const u8, mapping.size)
-                };
-                return Self::elf_file_identifier_from_mapped_file(mem_slice);
-            } else {
-                let mem_slice = Self::copy_from_process(
-                    pid,
-                    mapping.start_address as *mut libc::c_void,
-                    mapping.size,
-                )?;
-                return Self::elf_file_identifier_from_mapped_file(&mem_slice);
+        // FIXME `fixup_deleted_file` mutations of `mapping` were removed. Will this have a
+        // meaningful effect elsewhere?
```

Comment:
If the file was moved then the new code should be able to read from it and the fixup shouldn't be needed anymore. Similarly, if it's been deleted the kernel should have been holding on to its copy in memory. My guess is that we don't need to cover any of these cases explicitly anymore.

Reply:
I think maybe we still want to remove ` (deleted)` from names though? Otherwise the writer will write e.g. the main module with that suffix (as in the `with_deleted_binary` test where I commented out the assertion), which isn't what we want.